src/modules/folder/folder.usecase.ts

```diff
@@ -252,41 +262,55 @@ export class FolderUseCases {
       user = await this.userRepository.findByUsername(creator.bridgeUser);
     }
 
-    const parentFolderExists = await this.folderRepository.findOne({
-      id: parentFolderId,
+    const parentFolder = await this.folderRepository.findOne({
+      uuid: parentFolderUuid,
       userId: user.id,
     });
 
-    if (!parentFolderExists) {
-      throw new Error('Parent folder does not exist or is not yours');
+    if (!parentFolder) {
+      throw new InvalidParentFolderException(
+        'Parent folder does not exist or is not yours',
+      );
     }
 
     if (name === '' || invalidName.test(name)) {
-      throw new Error('Invalid folder name');
+      throw new BadRequestException('Invalid folder name');
     }
 
-    const encryptedFolderName = this.cryptoService.encryptName(
-      name,
-      parentFolderId,
-    );
-
     const nameAlreadyInUse = await this.folderRepository.findOne({
-      parentId: parentFolderId,
-      name: encryptedFolderName,
+      parentId: parentFolder.id,
+      plainName: name,
+      deleted: false,
     });
 
     if (nameAlreadyInUse) {
-      throw Error('Folder with the same name already exists');
+      throw new BadRequestException(
+        'Folder with the same name already exists in this location',
+      );
     }
 
-    const folder = await this.folderRepository.create(
-      user.id,
-      encryptedFolderName,
-      null,
-      parentFolderId,
-      '03-aes',
+    const encryptedFolderName = this.cryptoService.encryptName(
+      name,
+      parentFolder.id,
     );
 
+    const folder = await this.folderRepository.createWithAttributes({
+      uuid: v4(),
+      userId: user.id,
+      name: encryptedFolderName,
+      plainName: name,
+      parentId: parentFolder.id,
+      parentUuid: parentFolder.uuid,
+      encryptVersion: '03-aes',
+      bucket: null,
+      deleted: false,
+      removed: false,
+      createdAt: new Date(),
+      updatedAt: new Date(),
+      removedAt: null,
+      deletedAt: null,
+    });
+
     return folder;
   }
```

Comment:
This usecase is being used for nothing in this repository, so I adapted it to a closer (you could say is a clone) version of what we have in drive-server currently, the only difference is the use of uuid.